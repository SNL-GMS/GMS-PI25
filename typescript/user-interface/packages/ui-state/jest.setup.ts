/* eslint-disable max-classes-per-file */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable import/first */
/* eslint-disable import/no-extraneous-dependencies */
import 'fake-indexeddb/auto';

import type { FilterDefinition } from '@gms/common-model/lib/filter/types';
import crypto from 'crypto';
import * as util from 'util';

// This is to fix an issue with test leaks related to createStateSyncMiddleware() in jsdom
process.env.GMS_DISABLE_REDUX_STATE_SYNC = 'true';

Object.defineProperty(global, 'crypto', {
  value: {
    subtle: {
      digest: async (algorithm: AlgorithmIdentifier, msg: Uint8Array) =>
        (crypto.webcrypto as any).subtle.digest(algorithm, msg) as Promise<any>
    }
  }
});

Object.defineProperty(window, 'TextEncoder', {
  writable: true,
  value: util.TextEncoder
});
Object.defineProperty(window, 'TextDecoder', {
  writable: true,
  value: util.TextDecoder
});
Object.defineProperty(global, 'TextEncoder', {
  writable: true,
  value: util.TextEncoder
});
Object.defineProperty(global, 'TextDecoder', {
  writable: true,
  value: util.TextDecoder
});

import Adapter from '@cfaester/enzyme-adapter-react-18';
import Enzyme, { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import type { GlobalWithFetchMock } from 'jest-fetch-mock';

const customGlobal: GlobalWithFetchMock = (global as unknown) as GlobalWithFetchMock;
// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
customGlobal.fetch = require('jest-fetch-mock');

customGlobal.fetchMock = customGlobal.fetch;

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
require('jest-canvas-mock');

Enzyme.configure({ adapter: new Adapter() });

const globalAny: any = global;

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
globalAny.fetch = require('jest-fetch-mock');

// Make Enzyme functions available in all test files without importing
globalAny.shallow = shallow;
globalAny.render = render;
globalAny.mount = mount;
globalAny.toJson = toJson;
globalAny.TextEncoder = util.TextEncoder;

globalAny.window = window;

class Worker {
  public url: string;

  public onmessage: (message: Record<string, unknown>, transfer?: Transferable[]) => void;

  public constructor(stringUrl: string) {
    this.url = stringUrl;
    this.onmessage = () => {
      // do nothing
    };
  }

  public postMessage(message: Record<string, unknown>, transfer?: Transferable[]) {
    this.onmessage(message, transfer);
  }
}
globalAny.window.Worker = Worker;

class MessagePort {
  public url: string;

  public onmessage: (message: Record<string, unknown>, transfer?: Transferable[]) => void;

  public postMessage(message: Record<string, unknown>, transfer?: Transferable[]) {
    this.onmessage(message, transfer);
  }
}

class SharedWorker {
  public url: string;

  public port = new MessagePort();

  public constructor(stringUrl: string) {
    this.url = stringUrl;

    this.port.onmessage = () => {
      // do nothing
    };
  }
}
globalAny.window.SharedWorker = SharedWorker;

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports,  import/no-extraneous-dependencies
require('jest-fetch-mock').enableMocks();

jest.mock('@gms/ui-wasm', () => {
  const isDesigned = jest.requireActual('@gms/ui-wasm/lib/gms-filters-provider/is-designed');

  const filter = async (
    filterDefinition: FilterDefinition,
    data: Float64Array
  ): Promise<Float64Array> => Promise.resolve(data);

  const design = async (filterDefinition: FilterDefinition): Promise<FilterDefinition> =>
    Promise.resolve(filterDefinition);

  const getBoundsForPositionBuffer = async (): Promise<{
    min: number;
    minSecs: number;
    max: number;
    maxSecs: number;
  }> => Promise.resolve({ min: 0, minSecs: 0, max: 10, maxSecs: 10 });

  return {
    design,
    filter,
    getBoundsForPositionBuffer,
    isDesigned: isDesigned.isDesigned
  };
});
