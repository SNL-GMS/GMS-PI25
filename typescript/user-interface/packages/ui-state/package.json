{
  "name": "@gms/ui-state",
  "version": "0.1.0",
  "description": "GMS UI State",
  "license": "UNLICENSED",
  "author": "GMS Development Team",
  "sideEffects": false,
  "module": "./lib/ui-state.js",
  "typings": "./lib/ui-state.d.ts",
  "files": [
    "lib"
  ],
  "scripts": {
    "clean:build": "yarn rimraf lib/ dist/ docs/ coverage/ bundle-analyzer/ artifacts/ .cache/ junit.xml",
    "clean": "npm-run-all -s clean:build",
    "build:eslint": "yarn eslint -c .eslintrc-config.yaml 'src/**/*.{ts,tsx}' --quiet --cache --cache-strategy content --cache-file .cache/.eslintcache",
    "build:eslint:test": "yarn eslint -c .eslintrc-config-test.yaml '{__tests__,__mocks__,__integration-tests__}/**/*.{ts,tsx}' --quiet --cache --cache-strategy content --cache-file .cache/.eslintcache-test",
    "build": "yarn tsc -b ./tsconfig-build.json",
    "build:test": "yarn tsc -b ./tsconfig-test.json",
    "watch": "yarn tsc -b ./tsconfig-build.json -watch",
    "docs": "yarn clean:docs && typedoc --out docs ./src/ --name ui-state --mode file --readme README.md --target ES6 --includeDeclarations --excludeExternals --excludeNotExported --hideGenerator",
    "sonar": "echo \"run 'yarn sonar' from the top-level directory\"",
    "test": "yarn test:dev",
    "test:dev:jest": "jest --maxWorkers=50% --verbose --bail",
    "test:prod:jest": "jest --maxWorkers=4 --silent --bail --coverageReporters=lcovonly --reporters=default --reporters=jest-junit",
    "test:dev": "npm-run-all -p build:test test:dev:jest",
    "test:prod": "npm-run-all -p build:test test:prod:jest",
    "version": "echo $npm_package_version"
  },
  "dependencies": {
    "@blueprintjs/core": "4.19.2",
    "@gms/common-model": "0.1.0",
    "@gms/common-util": "0.1.0",
    "@gms/ui-util": "0.1.0",
    "@gms/ui-wasm": "0.1.0",
    "@gms/ui-workers": "0.1.0",
    "@gms/weavess-core": "0.1.0",
    "@reduxjs/toolkit": "1.9.1",
    "axios": "0.22.0",
    "d3": "5.5.0",
    "dexie": "3.2.2",
    "immer": "9.0.6",
    "immutable": "4.0.0-rc.12",
    "lodash": "4.17.21",
    "moment": "2.26.0",
    "msgpack-lite": "0.1.26",
    "p-queue": "6.6.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-redux": "8.0.5",
    "redux": "4.1.2",
    "redux-devtools-extension": "2.13.9",
    "redux-logger": "3.0.6",
    "redux-state-sync": "3.1.4",
    "redux-thunk": "2.4.1",
    "workbox-core": "6.5.4",
    "workbox-precaching": "6.5.4",
    "workbox-routing": "6.5.4",
    "workbox-strategies": "6.5.4",
    "worker-rpc": "0.2.0"
  },
  "devDependencies": {
    "@blueprintjs/eslint-plugin": "2.2.0",
    "@cfaester/enzyme-adapter-react-18": "0.6.0",
    "@testing-library/jest-dom": "5.16.5",
    "@testing-library/react": "13.4.0",
    "@testing-library/react-hooks": "7.0.2",
    "@types/faker": "4.1.8",
    "@types/jest": "29.1.1",
    "@types/lodash": "4.14.149",
    "@types/node": "18.11.9",
    "@types/react": "18.0.26",
    "@types/react-dom": "18.0.9",
    "@types/redux-logger": "3.0.9",
    "@types/redux-mock-store": "1.0.3",
    "@types/redux-state-sync": "3.1.3",
    "@typescript-eslint/eslint-plugin": "5.33.0",
    "@typescript-eslint/parser": "5.33.0",
    "enzyme": "3.11.0",
    "enzyme-to-json": "3.4.3",
    "eslint": "8.22.0",
    "eslint-config-airbnb": "19.0.4",
    "eslint-config-prettier": "8.5.0",
    "eslint-import-resolver-typescript": "3.4.1",
    "eslint-plugin-import": "2.26.0",
    "eslint-plugin-jest": "27.0.4",
    "eslint-plugin-jsdoc": "39.3.6",
    "eslint-plugin-jsx-a11y": "6.6.1",
    "eslint-plugin-no-null": "1.0.2",
    "eslint-plugin-prefer-arrow": "1.2.3",
    "eslint-plugin-prettier": "4.2.1",
    "eslint-plugin-react": "7.30.1",
    "eslint-plugin-react-hooks": "4.6.0",
    "eslint-plugin-simple-import-sort": "7.0.0",
    "eslint-plugin-unicorn": "43.0.2",
    "eslint-plugin-workspaces": "0.7.0",
    "fake-indexeddb": "4.0.0",
    "faker": "4.1.0",
    "immer": "9.0.6",
    "jest": "29.1.2",
    "jest-canvas-mock": "2.4.0",
    "jest-cli": "29.1.2",
    "jest-environment-jsdom": "29.1.2",
    "jest-fetch-mock": "3.0.3",
    "jest-junit": "12.2.0",
    "jsdom": "20.0.1",
    "jsdom-global": "3.0.2",
    "node-fetch": "2.6.0",
    "npm-run-all": "4.1.5",
    "prettier": "2.1.2",
    "react-test-renderer": "18.2.0",
    "redux-mock-store": "1.5.4",
    "rimraf": "3.0.0",
    "ts-jest": "29.1.1",
    "ts-jest-mock-import-meta": "0.12.0",
    "ts-node": "10.9.1",
    "typedoc": "0.15.3",
    "typescript": "5.1.6"
  },
  "jest-junit": {
    "suiteNameTemplate": "ui-state/{filename}",
    "classNameTemplate": "ui-state/{filename}/{classname}",
    "titleTemplate": "{title}"
  }
}
