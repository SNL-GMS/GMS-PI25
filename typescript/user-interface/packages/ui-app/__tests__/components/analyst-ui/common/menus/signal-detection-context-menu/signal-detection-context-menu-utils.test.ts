import type { SignalDetectionTypes } from '@gms/common-model';
import {
  deletedEventData,
  eventData,
  rejectedEventData,
  signalDetectionsData
} from '@gms/common-model/__tests__/__data__';
import { AnalystWorkspaceTypes } from '@gms/ui-state';

import {
  canDisplayFkForSds,
  canGenerateFk,
  getSignalDetectionDetailsProps,
  hideMeasurementModeEntries,
  showMeasurementModeEntries,
  toggleShownSDs
} from '~analyst-ui/common/menus/signal-detection-context-menu/signal-detection-context-menu-utils';

describe('Signal Detection Context Menu Utils', () => {
  const sdhypIds: string[] = [];
  signalDetectionsData.forEach(sd => {
    sd.signalDetectionHypotheses.forEach(sdhypo => {
      sdhypIds.push(sdhypo.id.id);
    });
  });

  const mockSetMeasurementModeEntries = jest.fn();
  const measurementMode = {
    mode: AnalystWorkspaceTypes.WaveformDisplayMode.MEASUREMENT,
    entries: {
      [signalDetectionsData[0].id]: true
    }
  };

  it('Should be defined', () => {
    expect(canDisplayFkForSds).toBeDefined();
    expect(canGenerateFk).toBeDefined();
    expect(getSignalDetectionDetailsProps).toBeDefined();
    expect(hideMeasurementModeEntries).toBeDefined();
    expect(showMeasurementModeEntries).toBeDefined();
    expect(toggleShownSDs).toBeDefined();
  });

  it('canGenerateFk', () => {
    const result: boolean = canGenerateFk(signalDetectionsData[0]);
    expect(result).toBe(true);
  });

  it('getSignalDetectionDetailsProps', () => {
    const mockUiThemeColors = {
      colors: {
        openEventSDColor: '#000000',
        completeEventSDColor: '#111111',
        otherEventSDColor: '#222222',
        unassociatedSDColor: '#333333'
      }
    };
    const sdDetailsProps = getSignalDetectionDetailsProps(
      signalDetectionsData[0],
      [eventData, deletedEventData, rejectedEventData],
      eventData.id,
      undefined,
      'AL1',
      { startTimeSecs: 1638297900, endTimeSecs: 1638298200 },
      mockUiThemeColors as any
    );
    expect(sdDetailsProps).toMatchSnapshot();
  });

  describe('canDisplayFkForSds', () => {
    it('canDisplayFkForSds is false', () => {
      const result: boolean = canDisplayFkForSds([] as SignalDetectionTypes.SignalDetection[]);
      expect(result).toBe(false);
    });
  });

  it('hideMeasurementModeEntries', () => {
    hideMeasurementModeEntries(
      sdhypIds,
      measurementMode,
      signalDetectionsData,
      mockSetMeasurementModeEntries
    );

    expect(mockSetMeasurementModeEntries).toHaveBeenCalledWith({
      '012de1b9-8ae3-3fd4-800d-58123c3152cc': false,
      '012de1b9-8ae3-3fd4-800d-58165c3152cc': false,
      '012de1b9-8ae3-3fd4-800d-58665c3152cc': false,
      '012de1b9-8ae3-3fd4-800d-58665c3152dd': false
    });
  });

  it('showMeasurementModeEntries', () => {
    showMeasurementModeEntries(
      sdhypIds,
      measurementMode,
      signalDetectionsData,
      mockSetMeasurementModeEntries
    );

    expect(mockSetMeasurementModeEntries).toHaveBeenCalledWith({
      '012de1b9-8ae3-3fd4-800d-58123c3152cc': true,
      '012de1b9-8ae3-3fd4-800d-58165c3152cc': true,
      '012de1b9-8ae3-3fd4-800d-58665c3152cc': true,
      '012de1b9-8ae3-3fd4-800d-58665c3152dd': true
    });
  });

  it('toggleShownSDs', () => {
    toggleShownSDs(
      signalDetectionsData.map(sd => sd.id),
      true,
      true,
      measurementMode,
      mockSetMeasurementModeEntries
    );

    expect(mockSetMeasurementModeEntries).toHaveBeenCalledWith({
      '012de1b9-8ae3-3fd4-800d-58123c3152cc': false,
      '012de1b9-8ae3-3fd4-800d-58165c3152cc': false,
      '012de1b9-8ae3-3fd4-800d-58665c3152cc': false,
      '012de1b9-8ae3-3fd4-800d-58665c3152dd': false
    });
  });
});
