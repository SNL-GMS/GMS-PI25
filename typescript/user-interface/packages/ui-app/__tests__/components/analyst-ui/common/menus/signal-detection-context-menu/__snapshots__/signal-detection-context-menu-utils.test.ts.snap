// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signal Detection Context Menu Utils getSignalDetectionDetailsProps 1`] = `
{
  "assocStatus": "Associated to open event",
  "color": "#000000",
  "signalDetection": {
    "id": "012de1b9-8ae3-3fd4-800d-58665c3152cc",
    "monitoringOrganization": "GMS",
    "signalDetectionHypotheses": [
      {
        "deleted": false,
        "featureMeasurements": [
          {
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "LONG_PERIOD_FIRST_MOTION",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "confidence": null,
              "referenceTime": 1636503404,
              "value": "INDETERMINATE",
            },
            "snr": null,
          },
          {
            "analysisWaveform": {
              "filterDefinition": undefined,
              "filterDefinitionUsage": undefined,
              "waveform": {
                "id": {
                  "channel": {
                    "effectiveAt": 1636503404,
                    "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                  },
                  "creationTime": 1636503404,
                  "endTime": 1636503704,
                  "startTime": 1636503404,
                },
              },
            },
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "ARRIVAL_TIME",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "arrivalTime": {
                "standardDeviation": 1.162,
                "value": 1636503404,
              },
              "travelTime": null,
            },
            "snr": {
              "standardDeviation": null,
              "units": "DECIBELS",
              "value": 8.9939442,
            },
          },
          {
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "PHASE",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "confidence": 50,
              "referenceTime": 1636503404,
              "value": "P",
            },
            "snr": null,
          },
          {
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "SHORT_PERIOD_FIRST_MOTION",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "confidence": null,
              "referenceTime": 1636503404,
              "value": "INDETERMINATE",
            },
            "snr": null,
          },
          {
            "analysisWaveform": {
              "filterDefinition": undefined,
              "filterDefinitionUsage": undefined,
              "waveform": {
                "id": {
                  "channel": {
                    "effectiveAt": 1636503404,
                    "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                  },
                  "creationTime": 1636503404,
                  "endTime": 1636503704,
                  "startTime": 1636503404,
                },
              },
            },
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "RECEIVER_TO_SOURCE_AZIMUTH",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "measuredValue": {
                "standardDeviation": 2.2089042,
                "units": "DEGREES",
                "value": 90.142121,
              },
              "referenceTime": null,
            },
            "snr": null,
          },
          {
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "EMERGENCE_ANGLE",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "measuredValue": {
                "standardDeviation": null,
                "units": "DEGREES",
                "value": -1,
              },
              "referenceTime": null,
            },
            "snr": null,
          },
          {
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "SLOWNESS",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "measuredValue": {
                "standardDeviation": 0.27,
                "units": "SECONDS_PER_DEGREE",
                "value": 7.1219535,
              },
              "referenceTime": null,
            },
            "snr": null,
          },
          {
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "RECTILINEARITY",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "measuredValue": {
                "standardDeviation": null,
                "units": "UNITLESS",
                "value": -1,
              },
              "referenceTime": null,
            },
            "snr": null,
          },
          {
            "channel": {
              "effectiveAt": 1636503404,
              "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
            },
            "featureMeasurementType": "AMPLITUDE_A5_OVER_2",
            "measuredChannelSegment": {
              "id": {
                "channel": {
                  "effectiveAt": 1636503404,
                  "name": "ASAR.beam.SHZ/beam,fk,coherent/steer,az_90.142deg,slow_7.122s_per_deg/06c0cb24-ab8f-3853-941d-bdf5e73a51b4",
                },
                "creationTime": 1636503404,
                "endTime": 1636503704,
                "startTime": 1636503404,
              },
            },
            "measurementValue": {
              "amplitude": {
                "standardDeviation": 0.5,
                "units": "MAGNITUDE",
                "value": 5,
              },
              "clipped": false,
              "measurementTime": 1636503404,
              "measurementWindowDuration": 6,
              "measurementWindowStart": 1636503404,
              "period": 0.25,
            },
            "snr": {
              "standardDeviation": null,
              "units": "DECIBELS",
              "value": 8.9939442,
            },
          },
        ],
        "id": {
          "id": "20cc9505-efe3-3068-b7d5-59196f37992c",
          "signalDetectionId": "012de1b9-8ae3-3fd4-800d-58665c3152cc",
        },
        "monitoringOrganization": "GMS",
        "parentSignalDetectionHypothesis": null,
        "station": {
          "effectiveAt": null,
          "name": "ASAR",
        },
      },
    ],
    "station": {
      "name": "ASAR",
    },
  },
  "swatchTooltipText": "Open event, before interval",
}
`;
